### ISSUE TITLE
Figwheel throws error when :warning-handlers are defined with inlined fn

### ISSUE THREAD

**Issue by Gonzih** [bhauman/lein-figwheel#262]
Hello,

I have testing compiler definition in my :cljsbuild -> :builds section that contains following section:

```clojure
:warning-handlers [(fn [warning-type env extra]
                      (when-let [s (cljs.analyzer/error-message warning-type extra)]
                        (throw (Exception. (str (cljs.analyzer/message env s))))))]}
```

This compilation target is not consumed by figwheel build directly (simple optimization mode to run inside v8 in ci).

Figwheel fails with error if this section is present in the configuration. Everything works fine only when (fn...) is removed.

Here is error message that figwheel throws:

```text
Exception in thread "main" java.lang.RuntimeException: Can't take value of a macro: #'clojure.core/fn, compiling:(/tmp/form-init1899398365858643590.c
lj:1:158)
```

Any tips on how to fix that?
Thanks a lot!

**Comment by bhauman** [comment]
Yet another problem with this stuff crossing the serialization boundary. I have no idea how cljsbuild manages this so effortlessly.

I wonder if you can place this function in user.clj and then refer to it? `:warning-handlers [#'myfun]`

Other than you can place figwheel only builds into `:figwheel {:builds []}`. And then figwheel won't read the cljsbuilds.

**Comment by Gonzih** [comment]
Solved by putting build configuration in to :figwheel {:builds []} section.

Thanks!

I was suspecting that it's a problem with serialization.

**Comment by bhauman** [comment]
Alright I see this is a new cljsbuild feature that I didn't know about.  I will have to filter it out.

**Comment by Gonzih** [comment]
Thanks for your time!

**Comment by danielcompton** [comment]
I'm getting a similar issue setting `:closure-defines in my project.clj. They are handled fine by cljsbuild, but when I try to run figwheel on that project.clj (even if they're not in the project I'm trying to build), then it throws an exception.

**Comment by bhauman** [comment]
if you use strings for your keys and not quotes it should work fine

**Comment by bhauman** [comment]
Actually I just committed a fix for this. This will be available in the 0.5.0 release of figwheel.
